consul/src/main/java/com/linecorp/armeria/server/consul/ConsulUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.internal.consul.ConsulClient;
+
+/**
+ * Builds a new {@link ConsulUpdatingListener}, which registers the server to a Consul cluster.
+ * <h2>Examples</h2>
+ * <pre>{@code
+ * ConsulUpdatingListener listener =
+ *     new ConsulUpdatingListenerBuilder("myService").url("http://myConsulHost:8500").build();
+ * ServerBuilder sb = Server.builder();
+ * sb.addListener(listener);
+ * }</pre>
+ */
+public final class ConsulUpdatingListenerBuilder {
+    private final String serviceName;
+    @Nullable private String consulUrl;
+    @Nullable private Endpoint serviceEndpoint;
+
+    @Nullable private String checkUrl;
+    @Nullable private String checkMethod;
+    @Nullable private String checkInterval;
+
+    /**
+     * Creates a {@link ConsulUpdatingListenerBuilder} with a service name.
+     *
+     * @param serviceName Service name to register
+     */
+    public ConsulUpdatingListenerBuilder(String serviceName) {
```

Comment:
The constructor of this builder should be package-private so that users cannot create this class directly.
```suggestion
    ConsulUpdatingListenerBuilder(String serviceName) {
```